deep_qa/models/sentence_selection/siamese_sentence_selector.py

```diff
@@ -0,0 +1,123 @@
+
+        # shape: (batch size, num_sentences, sentences_length, embedding size)
+        sentences_embedding = self._embed_input(sentences_input)
+
+        # We encode the question embeddings with some encoder.
+        question_encoder = self._get_encoder(name="question_encoder",
+                                             fallback_behavior="use default encoder")
+        # shape: (batch size, encoder_output_dimension)
+        encoded_question = question_encoder(question_embedding)
+
+        # We encode the document embeddings with some encoder.
+        sentences_encoder = EncoderWrapper(self._get_encoder(name="sentence_encoder",
+                                                             fallback_behavior="use default encoder"),
+                                           name="TimeDistributed_sentences_encoder")
+        # shape: (batch size, num_sentences, encoder_output_dimension)
+        encoded_sentences = sentences_encoder(sentences_embedding)
+
+        # Here we use the Attention layer with the cosine similarity function
+        # to get the cosine similarities of each sesntence with the question.
+        # shape: (batch size, num_sentences)
+        attention_name = 'question_sentences_similarity'
+        similarity_params = {"type": "cosine_similarity"}
+        sentence_probabilities = Attention(name=attention_name,
+                                           similarity_function=similarity_params)([encoded_question,
+                                                                                   encoded_sentences])
+
+        return DeepQaModel(input=[question_input, sentences_input],
+                           output=sentence_probabilities)
+
+    @overrides
+    def _instance_type(self):
+        """
+        Return the instance type that the model trains on.
+        """
+        return SentenceSelectionInstance
+
+    @overrides
+    def _get_max_lengths(self) -> Dict[str, int]:
+        """
+        Return a dictionary with the appropriate padding lengths.
+        """
+        max_lengths = super(SiameseSentenceSelector, self)._get_max_lengths()
+        max_lengths['num_question_words'] = self.num_question_words
+        max_lengths['num_sentences'] = self.num_sentences
+        return max_lengths
+
+    @overrides
+    def _set_max_lengths(self, max_lengths: Dict[str, int]):
+        """
+        Set the padding lengths of the model.
+        """
+        super(SiameseSentenceSelector, self)._set_max_lengths(max_lengths)
+        self.num_question_words = max_lengths['num_question_words']
+        self.num_sentences = max_lengths['num_sentences']
+
+    @overrides
+    def _set_max_lengths_from_model(self):
+        self.set_text_lengths_from_model_input(self.model.get_input_shape_at(0)[1][2:])
+        self.num_question_words = self.model.get_input_shape_at(0)[0][1]
+        self.num_sentences = self.model.get_input_shape_at(0)[1][1]
+        self.num_sentence_words = self.model.get_input_shape_at(0)[1][2]
```

Comment:
This gets handled with the call to `self.set_text_lengths_from_model_input` - no need to have this line here.